lib/tpm_hash.c

```diff
@@ -97,3 +97,52 @@ int tpm_hash_compute_data(TSS2_SYS_CONTEXT *sapi_context, BYTE *buffer,
     free(bufferList);
     return rval == TPM_RC_SUCCESS ? 0 : -3;
 }
+
+//
+// This function does a hash on an array of data strings and re-uses syscontext
+//
+UINT32 tpm_hash_sequence( TSS2_SYS_CONTEXT *sysContext, TPMI_ALG_HASH hashAlg,
+    UINT8 numBuffers, TPM2B_DIGEST *bufferList, TPM2B_DIGEST *result ) {
+
+    TPMT_TK_HASHCHECK validation;
+    TPMS_AUTH_COMMAND cmdAuth;
+    TPMS_AUTH_COMMAND *cmdSessionArray[1] = { &cmdAuth };
+    TSS2_SYS_CMD_AUTHS cmdAuthArray = { 1, &cmdSessionArray[0] };
+    TPMI_DH_OBJECT sequenceHandle;
+    TPM2B_AUTH nullAuth = {
+        .t.size = 0
+    };
+    TPM2B emptyBuffer = {
+        .size = 0
+    };
+
+    // Set result size to 0, in case any errors occur
+    result->b.size = 0;
+    
+    // Init input sessions struct
+    cmdAuth.sessionHandle = TPM_RS_PW;
+    cmdAuth.nonce.t.size = 0;
+    *( (UINT8 *)((void *)&cmdAuth.sessionAttributes ) ) = 0;
```

Comment:
You don't need this cmdAuth is now zeroed out, I know this was a copy + paste, but this is also the wrong way to do this. It assumed that the sessionAttributes bit field would never change and only set the 8 defined values instead of just setting sessionAttributes to zero. Also, on borland compilers, bit fields start in the other direction, so this would be broken on that compiler.